ts/screens/preferences/CalendarScreen.tsx

```diff
@@ -0,0 +1,224 @@
+};
+
+type ResourceError = FetchError;
+
+const mapResourceErrorToMessage = (resourceError: ResourceError): string => {
+  switch (resourceError.kind) {
+    case "FETCH_ERROR":
+      return I18n.t("messages.cta.errors.fetchCalendars");
+
+    default: {
+      // Exhaustive check
+      return ((): never => resourceError.kind)();
+    }
+  }
+};
+
+/**
+ * Implements the biometric recognition preference screen where the user can
+ * opt for explicitly not using fingerprint. This class shows up two possible
+ * scenarios:
+ * * a switch to enable/disable biometric recognition. Enabled if the
+ *   fingerprint is enrolled.
+ * * the above switch (disabled) with a button below that redirects the user to
+ *   device settings otherwise.
+ *
+ * Please note that this screen must not be displayed if biometric recognition
+ * is unavailable.
+ */
+/**
+ * A modal that allow the user to select one of the device available Calendars
+ */
+class CalendarScreen extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = INITIAL_STATE;
+  }
+
+  private goBack = () => this.props.navigation.goBack();
+
+  private onCalendarSelected = (calendar: Calendar) => {
+    this.props.preferredCalendarSaveSuccess(calendar);
+
+    this.goBack();
+  };
+
+  public render() {
+    const { calendars } = this.state;
+
+    return (
+      <TopScreenComponent
+        headerTitle={I18n.t("preferences.title")}
+        title={I18n.t("preferences.list.preferred_calendar.title")}
+        goBack={this.goBack}
+        subtitle={I18n.t("messages.cta.reminderCalendarSelect")}
+      >
+        <View
+          style={{
+            width: "100%",
+            flexDirection: "row",
+            alignItems: "center",
+            justifyContent: "space-between",
+            paddingHorizontal: 25
+          }}
+        />
+        <View style={styles.content}>
+          {pot.isLoading(calendars) && <Text>Loading calendars...</Text>}
+          {pot.isError(calendars) && (
+            <React.Fragment>
+              <Text>{mapResourceErrorToMessage(calendars.error)}</Text>
+              <Button onPress={this.fetchCalendars}>
+                <Text>{I18n.t("global.buttons.retry")}</Text>
+              </Button>
+            </React.Fragment>
+          )}
+          {pot.isSome(calendars) && (
+            <React.Fragment>
+              {calendars.value.map(calendar => (
+                <CalendarItem
+                  key={calendar.id}
+                  calendar={calendar}
+                  onPress={() => this.onCalendarSelected(calendar)}
+                />
+              ))}
+            </React.Fragment>
+          )}
+          <Text style={styles.separator}>{I18n.t("messages.cta.helper")}</Text>
+        </View>
+      </TopScreenComponent>
+    );
+  }
+
+  public componentDidMount() {
+    this.fetchCalendars();
+  }
+
+  private fetchCalendars = () => {
+    this.setState({ calendars: pot.toLoading(pot.none) });
```

Comment:
```suggestion
    this.setState({ calendars: pot.noneLoading });
```

`toLoading` can be used to convert a previous pot value to loading

Reply:
changed